internal/service/ivschat/logging_configuration.go

```diff
@@ -0,0 +1,426 @@
+		Identifier: aws.String(d.Id()),
+	})
+
+	if err != nil {
+		var nfe *types.ResourceNotFoundException
+		if errors.As(err, &nfe) {
+			return nil
+		}
+
+		return create.DiagError(names.IVSChat, create.ErrActionDeleting, ResNameLoggingConfiguration, d.Id(), err)
+	}
+
+	if _, err := waitLoggingConfigurationDeleted(ctx, conn, d.Id(), d.Timeout(schema.TimeoutDelete)); err != nil {
+		return create.DiagError(names.IVSChat, create.ErrActionWaitingForDeletion, ResNameLoggingConfiguration, d.Id(), err)
+	}
+
+	return nil
+}
+
+func flattenDestinationConfiguration(apiObject types.DestinationConfiguration) []interface{} {
+	if apiObject == nil {
+		return []interface{}{}
+	}
+
+	m := map[string]interface{}{}
+
+	switch v := apiObject.(type) {
+	case *types.DestinationConfigurationMemberCloudWatchLogs:
+		m["cloudwatch_logs"] = flattenCloudWatchDestinationConfiguration(v.Value)
+
+	case *types.DestinationConfigurationMemberFirehose:
+		m["firehose"] = flattenFirehoseDestinationConfiguration(v.Value)
+
+	case *types.DestinationConfigurationMemberS3:
+		m["s3"] = flattenS3DestinationConfiguration(v.Value)
+
+	case *types.UnknownUnionMember:
+		log.Println("unknown tag:", v.Tag)
+
+	default:
+		log.Println("union is nil or unknown type")
+	}
+
+	return []interface{}{m}
+}
+
+func flattenCloudWatchDestinationConfiguration(apiObject types.CloudWatchLogsDestinationConfiguration) []interface{} {
+	m := map[string]interface{}{}
+
+	if v := apiObject.LogGroupName; v != nil {
+		m["log_group_name"] = aws.ToString(v)
+	}
+
+	return []interface{}{m}
+}
+
+func flattenFirehoseDestinationConfiguration(apiObject types.FirehoseDestinationConfiguration) []interface{} {
+	m := map[string]interface{}{}
+
+	if v := apiObject.DeliveryStreamName; v != nil {
+		m["delivery_stream_name"] = aws.ToString(v)
+	}
+
+	return []interface{}{m}
+}
+
+func flattenS3DestinationConfiguration(apiObject types.S3DestinationConfiguration) []interface{} {
+	m := map[string]interface{}{}
+
+	if v := apiObject.BucketName; v != nil {
+		m["bucket_name"] = aws.ToString(v)
+	}
+
+	return []interface{}{m}
+}
+
+func expandDestinationConfiguration(vSettings []interface{}) types.DestinationConfiguration {
+	if len(vSettings) == 0 || vSettings[0] == nil {
+		return nil
+	}
+
+	tfMap := vSettings[0].(map[string]interface{})
+
+	if v, ok := tfMap["cloudwatch_logs"].([]interface{}); ok && len(v) > 0 {
+		return &types.DestinationConfigurationMemberCloudWatchLogs{
+			Value: *expandCloudWatchLogsDestinationConfiguration(v),
+		}
+	} else if v, ok := tfMap["firehose"].([]interface{}); ok && len(v) > 0 {
+		return &types.DestinationConfigurationMemberFirehose{
+			Value: *expandFirehouseDestinationConfiguration(v),
+		}
+	} else if v, ok := tfMap["s3"].([]interface{}); ok && len(v) > 0 {
+		return &types.DestinationConfigurationMemberS3{
+			Value: *expandS3DestinationConfiguration(v),
+		}
+	} else {
+		return nil
+	}
+}
```

Comment:
This pattern was taken from https://github.com/hashicorp/terraform-provider-aws/blob/d1e68d3f2987a8e75401d39df7b226ebb3d7c5b2/internal/service/identitystore/flex.go#L126-L142